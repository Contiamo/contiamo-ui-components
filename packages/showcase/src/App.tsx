import * as React from "react"
import { BrowserRouter as Router, Route, withRouter } from "react-router-dom"
import glamorous, { Div, ThemeProvider } from "glamorous"
import { css } from "glamor"
import { contiamoTheme } from "contiamo-ui-components"

import { appFontFace, appFontWeights, getFontSrcString } from "./utils/fonts"

import SideNavigation from "./components/SideNavigation/SideNavigation"
import Header from "./components/Header/Header"

import IntroPage from "./pages/Intro"
import ComponentsPage from "./pages/Components"

const Container = glamorous.div({
  display: "flex",
  flexDirection: "column",
  width: "100vw",
  backgroundColor: contiamoTheme.colors.usage.bodyBackground,
  height: "100vh",
  marginLeft: 60
})

const styles: {} = {
  display: "flex",
  fontFamily: "'Proxima Nova', sans-serif",
  "& hr": {
    margin: `${contiamoTheme.spacing * 3}px 0 ${contiamoTheme.spacing * 2}px`,
    height: 1,
    border: 0,
    backgroundColor: "rgba(0, 0, 0, 0.14)"
  },
  "& h1": {
    margin: 0,
    fontSize: "2rem"
  },
  "& h2": {
    fontSize: "1.3rem"
  },
  "& p": {
    margin: 0,
    marginBottom: contiamoTheme.spacing
  },
}

const SideNavigationWithRouter = withRouter(SideNavigation)

const App = () => (
  <Router>
    <ThemeProvider theme={contiamoTheme}>
      <div>
        <SideNavigationWithRouter />
        <Container>
          <Header />
          <Route exact path="/" component={IntroPage} />
          <Route path="/components" component={ComponentsPage} />
        </Container>
      </div>
    </ThemeProvider>
  </Router>
)

// @todo -> type this better
const x = css as any
appFontWeights.forEach((fontWeight: number) => {
  x.fontFace({
    ...appFontFace,
    ...getFontSrcString(fontWeight)()(),
    fontWeight
  })
})
x.global("body", { backgroundColor: contiamoTheme.colors.usage.bodyBackground })
x.global({
  fontSize: 13
})

export default glamorous(App)(styles)
