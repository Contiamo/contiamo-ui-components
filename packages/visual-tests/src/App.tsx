import * as React from "react"
import glamorous from "glamorous"
import { Sunburst, VisualizationWrapper } from "@operational/visualizations"
import { operational, Theme } from "@operational/theme"

import {
  OperationalUI,
  Button,
  Icon,
  Card,
  CardHeader,
  Grid,
  Sidebar,
  SidebarHeader,
  Small,
  SidebarItem,
} from "@operational/components"

import { darken } from "@operational/utils"

import Marathon, { MarathonEnvironment } from "./Marathon"
import MarathonRenderer from "./MarathonRenderer"
import allTestCases, { fromHash, toHash } from "./TestCases"

export interface State {
  group: number
  test: number
  isLooping: boolean
  isIdle: boolean
}

const TestToggle = glamorous.span(
  ({ theme, active }: { theme: Theme; active: boolean }): {} => ({
    display: "inline-block",
    marginRight: 16,
    padding: "2px 4px",
    cursor: "pointer",
    borderRadius: active ? 2 : 0,
    color: active ? theme.colors.white : theme.colors.linkText,
    backgroundColor: active ? theme.colors.linkText : theme.colors.white,
    borderBottom: "1px solid currentColor",
  }),
)

class App extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props)
    const indicesFromHash = fromHash(window.location.hash)(allTestCases)
    this.state = {
      group: indicesFromHash ? indicesFromHash.groupIndex : 0,
      test: indicesFromHash ? indicesFromHash.testIndex : 0,
      isLooping: false,
      isIdle: false,
    }
  }

  componentDidMount() {
    window.addEventListener("popstate", () => {
      this.readRoute()
    })
  }

  componentDidUpdate() {
    this.syncRoute()
  }

  readRoute() {
    const indicesFromHash = fromHash(window.location.hash)(allTestCases)
    if (indicesFromHash === null) {
      return
    }
    this.setState(() => ({
      group: indicesFromHash.groupIndex,
      test: indicesFromHash.testIndex,
    }))
  }

  syncRoute() {
    const hash = toHash({
      groupIndex: this.state.group,
      testIndex: this.state.test,
    })(allTestCases)
    if (hash !== window.location.hash) {
      history.pushState(null, null, hash)
    }
  }

  loop() {
    const reachedEnd = this.state.test >= allTestCases[this.state.group].children.length - 1
    setTimeout(() => {
      this.setState(prevState => ({
        test: reachedEnd ? 0 : prevState.test + 1,
      }))
    }, 2000)
  }

  render() {
    const test = allTestCases[this.state.group].children[this.state.test].marathon
    return (
      <OperationalUI withBaseStyles>
        <React.Fragment>
          <Grid type="IDE">
            <Card>
              <CardHeader>
                Tests
                <Button
                  condensed
                  color={this.state.isLooping ? "white" : "info"}
                  css={{ marginRight: 0 }}
                  onClick={() => {
                    if (!this.state.isLooping && this.state.isIdle) {
                      this.loop()
                    }
                    this.setState(prevState => ({
                      isLooping: !prevState.isLooping,
                      isIdle: !prevState.isLooping ? false : prevState.isIdle,
                    }))
                  }}
                >
                  {this.state.isLooping ? "Pause" : "Loop"}
                </Button>
              </CardHeader>
              <Sidebar
                css={{
                  margin: -operational.spacing,
                  width: `calc(100% + ${2 * operational.spacing}px)`,
                  boxShadow: "none",
                  borderTop: "1px solid #f2f2f2",
                }}
              >
                {allTestCases.map((test, groupIndex) => (
                  <SidebarHeader
                    key={groupIndex}
                    label={test.title}
                    open={groupIndex === this.state.group}
                    onToggle={() => {
                      this.setState(() => ({
                        group: groupIndex,
                        test: 0,
                      }))
                    }}
                  >
                    {test.children.map((test, testIndex) => (
                      <SidebarItem
                        key={testIndex}
                        active={groupIndex === this.state.group && testIndex === this.state.test}
                        onClick={() => {
                          this.setState(() => ({
                            test: testIndex,
                          }))
                        }}
                      >
                        {test.title}
                      </SidebarItem>
                    ))}
                  </SidebarHeader>
                ))}
              </Sidebar>
            </Card>
            <Card>
              <CardHeader>
                Canvas
                <a
                  href={`https://github.com/contiamo/operational-ui/tree/master/packages/visual-tests/src/TestCases/${
                    allTestCases[this.state.group].folder
                  }/${allTestCases[this.state.group].children[this.state.test].slug}.ts`}
                >
                  <Small>View code for this test</Small>
                </a>
              </CardHeader>
              <Marathon
                test={test}
                onCompleted={() => {
                  if (!this.state.isLooping && !this.state.isIdle) {
                    this.setState(prevState => ({
                      isIdle: true,
                    }))
                    return
                  }
                  if (this.state.isLooping) {
                    this.loop()
                  }
                }}
                timeout={2000}
              >
                {MarathonRenderer}
              </Marathon>
            </Card>
          </Grid>
        </React.Fragment>
      </OperationalUI>
    )
  }
}

export default App
