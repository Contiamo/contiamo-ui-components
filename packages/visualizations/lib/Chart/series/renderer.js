"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var area_1 = require("./renderers/area");
var bars_1 = require("./renderers/bars");
var flag_1 = require("./renderers/flag");
var line_1 = require("./renderers/line");
var symbol_1 = require("./renderers/symbol");
var text_1 = require("./renderers/text");
var Renderer = /** @class */ (function () {
    function Renderer(state, events, el, data, options, series) {
        switch (options.type) {
            case "area":
                return new area_1.default(state, events, el.select("g.series-area"), data, options, series);
            case "bars":
                return new bars_1.default(state, events, el.select("g.series-bars"), data, options, series);
            case "flag":
                return new flag_1.default(state, events, el.select("g.series-flag"), data, options, series);
            case "line":
                return new line_1.default(state, events, el.select("g.series-line"), data, options, series);
            case "symbol":
                return new symbol_1.default(state, events, el.select("g.series-symbol"), data, options, series);
            case "text":
                return new text_1.default(state, events, el.select("g.series-text"), data, options, series);
        }
    }
    return Renderer;
}());
exports.default = Renderer;
//# sourceMappingURL=renderer.js.map