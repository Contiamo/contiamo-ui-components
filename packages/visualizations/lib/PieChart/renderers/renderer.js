"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var donut_1 = require("./donut");
var polar_1 = require("./polar");
var gauge_1 = require("./gauge");
// Factory Class
var Renderer = /** @class */ (function () {
    function Renderer(state, events, el, options) {
        switch (options.type) {
            case "donut":
                return new donut_1.default(state, events, el, options);
            case "polar":
                return new polar_1.default(state, events, el, options);
            case "gauge":
                return new gauge_1.default(state, events, el, options);
            default:
                throw new Error("invalid render type '" + options.type + "' specified");
        }
    }
    return Renderer;
}());
exports.default = Renderer;
//# sourceMappingURL=renderer.js.map