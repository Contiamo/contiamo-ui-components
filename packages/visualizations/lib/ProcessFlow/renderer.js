"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var nodes_1 = require("./renderers/nodes");
var links_1 = require("./renderers/links");
var event_catalog_1 = require("../utils/event_catalog");
var fp_1 = require("lodash/fp");
var styles = require("./renderers/styles");
var d3_utils_1 = require("../utils/d3_utils");
var d3 = require("d3-selection");
var Renderer = /** @class */ (function () {
    function Renderer(state, events, el) {
        this.events = events;
        this.el = el;
        this.links = new links_1.default(state, events, el);
        this.nodes = new nodes_1.default(state, events, el);
        this.events.on(event_catalog_1.default.FOCUS.ELEMENT.HIGHLIGHT, this.focusElement.bind(this));
    }
    Renderer.prototype.draw = function (data) {
        this.links.draw(data.links);
        this.nodes.draw(data.nodes);
    };
    Renderer.prototype.focusElement = function (focusElement) {
        switch (focusElement.type) {
            case "path":
                this.highlightPath(focusElement);
                break;
            case "node":
                this.nodes.focusElement(focusElement);
                break;
            case "link":
                this.links.focusElement(focusElement);
                break;
        }
    };
    Renderer.prototype.highlightPath = function (focusElement) {
        var _this = this;
        this.events.emit(event_catalog_1.default.FOCUS.ELEMENT.OUT);
        var path = focusElement.matchers.path;
        var links = fp_1.zip(fp_1.initial(path))(fp_1.tail(path));
        fp_1.forEach(function (link) {
            _this.el
                .selectAll("path.link." + styles.element)
                .filter(function (d) { return d.sourceId() === link[0] && d.targetId() === link[1]; })
                .each(d3_utils_1.withD3Element(function (d, el) {
                _this.links.highlight(d3.select(el), d, true);
            }));
        })(links);
    };
    Renderer.prototype.close = function () {
        this.el.node().innerHTML = "";
    };
    return Renderer;
}());
exports.default = Renderer;
//# sourceMappingURL=renderer.js.map