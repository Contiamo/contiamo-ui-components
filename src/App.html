<!doctype html>
<html>
  <head>
    <title>Contiamo UI Components</title>
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>
