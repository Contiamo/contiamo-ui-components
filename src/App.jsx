// @flow
import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import glamorous, { Img, Div, ThemeProvider } from 'glamorous';
import { Box, Camera } from 'react-feather';

import { darken } from './utils/color';

import SideNavigation, {
  SideNavigationItem,
  SideNavigationLink,
} from './components/SideNavigation/SideNavigation';

import Header, { HeaderItem, HeaderSeparator, HeaderTitle } from './components/Header/Header';

import Sidebar, { SidebarItem, SidebarLink } from './components/Sidebar/Sidebar';

import Canvas from './Canvas/Canvas';

import logo from './img/logo/outline.png';

import DEFAULT_THEME from './App.theme';

/* @todo remove this after the sidebar is proper.
This just generates some sidebar items to play with. */
const getItems = [...Array(100)];

const App = ({ className, match }: { className: string }) =>
  (<Router>
    <ThemeProvider theme={DEFAULT_THEME}>
      <div className={className}>
        <SideNavigation color="#445873">
          <SideNavigationItem tooltip="Contiamo" size={60}>
            <Img css={{ maxWidth: '100%' }} alt="Contiamo" src={logo} />
          </SideNavigationItem>

          <SideNavigationItem tooltip="Components">
            <Box color="white" size={30} />
          </SideNavigationItem>

          {/* A simple separator */}
          <Div css={{ flexGrow: 1, height: '100%' }} />

          <SideNavigationItem
            tooltip={
              <div>
                <SideNavigationLink>My Account</SideNavigationLink>
                <SideNavigationLink>Feedback</SideNavigationLink>
                <SideNavigationLink>Logout</SideNavigationLink>
              </div>
            }
            tooltipAnchor="bottom"
            size={40}
          >
            <Img
              css={{ maxWidth: '100%' }}
              alt="Avatar"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRP0K2M10ttKd6aRHT-7LUhVB3rW2rVZ16N3yJux4pRFXd9jdWNi4eczg"
            />
          </SideNavigationItem>
        </SideNavigation>

        <Div css={{ display: 'flex', flexDirection: 'column', width: '100vw', height: '100vh' }}>
          <Header color="#445873">
            <HeaderTitle>Contiamo</HeaderTitle>
            <HeaderItem>
              <Camera /> Gallery
            </HeaderItem>
            <HeaderItem>
              <Box /> Components
            </HeaderItem>
            <HeaderSeparator />
            <HeaderItem>Logout</HeaderItem>
          </Header>

          <Canvas>
            <Route default>
              <Sidebar>
                {/* It's play time! See line 22 */}
                {getItems.map((item, index) =>
                  (<SidebarItem key={index} title="My">
                    <SidebarLink
                      tooltip="The Channel Group associated with an end user's session for this View (defined by the View's Channel Groupings)."
                      symbol="%"
                    >
                      Hola Compadre
                    </SidebarLink>
                    <SidebarLink to="/route" symbol="%">
                      No albo espa√±ol?
                    </SidebarLink>
                    <SidebarItem tooltip="Sup" key={index} title="My">
                      <SidebarLink tooltip="This means hi friend in spanish" symbol="%">
                        Hola Compadre
                      </SidebarLink>
                      <SidebarItem
                        onClick={() => new Promise(resolve => setTimeout(() => resolve(), 2000))}
                        key={index}
                        title="My"
                      >
                        <SidebarLink symbol="%">Hola Compadre</SidebarLink>
                        <SidebarItem key={index} title="My">
                          <SidebarLink symbol="%">Hola Compadre</SidebarLink>
                          <SidebarItem key={index} title="My">
                            <SidebarLink
                              tooltip={<img src="http://placehold.it/200x200" />}
                              symbol="%"
                            >
                              Hola Compadre
                            </SidebarLink>
                            <SidebarItem key={index} title="My">
                              <SidebarLink symbol="%">Hola Compadre</SidebarLink>
                            </SidebarItem>
                          </SidebarItem>
                        </SidebarItem>
                      </SidebarItem>
                    </SidebarItem>
                  </SidebarItem>),
                )}
              </Sidebar>
            </Route>
          </Canvas>
        </Div>
      </div>
    </ThemeProvider>
  </Router>);

const styles: {} = {
  display: 'flex',
  fontSize: DEFAULT_THEME.fonts.defaultSize,
  fontFamily: DEFAULT_THEME.fonts.main,
};

export default glamorous(App)(styles);
