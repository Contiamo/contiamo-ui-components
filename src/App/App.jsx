// @flow
import React from "react"
import { BrowserRouter as Router, Route, withRouter } from "react-router-dom"
import glamorous, { Div, ThemeProvider } from "glamorous"
import { css } from "glamor"

import { appFontFace, appFontWeights, getFontSrcString } from "./utils/fonts"

import SideNavigation from "./components/SideNavigation/SideNavigation"
import Header from "./components/Header/Header"
import Sidebar from "./components/Sidebar/Sidebar"
import AppCanvas from "./components/Canvas/Canvas"

import ButtonsPage from "./pages/Buttons/Buttons"
import FormFieldsPage from "./pages/FormFields/FormFields"
import CardsPage from "./pages/Cards/Cards"
import ChipsPage from "./pages/Chips/Chips"
import TooltipsPage from "./pages/Tooltips/Tooltips"
import StatsPage from "./pages/Stats/Stats"
import SidebarPage from "./pages/Sidebar/Sidebar"

import DEFAULT_THEME from "../theme"

const SidebarWithRouter = withRouter(Sidebar),
  App = ({ className }: { className: string }) =>
    <Router>
      <ThemeProvider theme={DEFAULT_THEME}>
        <div className={className}>
          <SideNavigation />
          <Div
            css={{
              display: "flex",
              flexDirection: "column",
              width: "100vw",
              height: "100vh"
            }}
          >
            <Header />
            <Div
              css={{
                display: "flex",
                padding: 16,
                width: "100%",
                height: "100vh"
              }}
            >
              <SidebarWithRouter />
              <AppCanvas css={{ marginLeft: 16, flexBasis: "100%" }}>
                <Route path="/buttons" component={ButtonsPage} />
                <Route path="/form-fields" component={FormFieldsPage} />
                <Route path="/cards" component={CardsPage} />
                <Route path="/chips" component={ChipsPage} />
                <Route path="/tooltips" component={TooltipsPage} />
                <Route path="/stats" component={StatsPage} />
                <Route path="/sidebar" component={SidebarPage} />
                <Route
                  path="/demo"
                  render={() =>
                    <img style={{ maxWidth: "175%" }} src="/screen.png" />}
                />
              </AppCanvas>
            </Div>
          </Div>
        </div>
      </ThemeProvider>
    </Router>,
  styles: {} = {
    display: "flex",
    backgroundColor: DEFAULT_THEME.greys["10"],
    ...DEFAULT_THEME.fonts,
    "& hr": {
      margin: `${DEFAULT_THEME.spacing * 3}px 0 ${DEFAULT_THEME.spacing * 2}px`,
      height: 1,
      border: 0,
      backgroundColor: "rgba(0, 0, 0, 0.14)"
    }
  }

appFontWeights.forEach((fontWeight: number) => {
  css.fontFace({
    ...appFontFace,
    ...getFontSrcString(fontWeight)()(),
    fontWeight
  })
})

export default glamorous(App)(styles)
