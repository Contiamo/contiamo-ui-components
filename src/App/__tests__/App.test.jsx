// @flow
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'

test('App starts', () => {
  const output: Object = renderer.create(<App />)
  const tree: {} = output.toJSON()
  expect(tree).toMatchSnapshot()
})
