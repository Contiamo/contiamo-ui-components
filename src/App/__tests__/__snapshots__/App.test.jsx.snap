// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App starts 1`] = `
<div
  className="css-1kcnbix"
>
  <div
    className="css-tacsng"
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="css-f02nqw"
        onClick={undefined}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Contiamo"
            className="css-1mjw833"
            src="img/logo/outline.png"
          />
        </a>
        
      </div>
    </div>
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="css-1uw152a"
        onClick={undefined}
      >
        <svg
          height={30}
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.89 1.45l8 4A2 2 0 0 1 22 7.24v9.53a2 2 0 0 1-1.11 1.79l-8 4a2 2 0 0 1-1.79 0l-8-4a2 2 0 0 1-1.1-1.8V7.24a2 2 0 0 1 1.11-1.79l8-4a2 2 0 0 1 1.78 0z"
            fill="none"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <polyline
            fill="none"
            points="2.32 6.16 12 11 21.68 6.16"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <line
            fill="none"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            x1="12"
            x2="12"
            y1="22.76"
            y2="11"
          />
        </svg>
        
      </div>
    </div>
    <div
      className="css-1c5lpzn"
    />
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="css-owk2pz"
        onClick={undefined}
      >
        <img
          alt="Avatar"
          className="css-1mjw833"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRP0K2M10ttKd6aRHT-7LUhVB3rW2rVZ16N3yJux4pRFXd9jdWNi4eczg"
        />
        
      </div>
    </div>
  </div>
  <div
    className="css-19imckl"
  >
    <div
      className="css-czadd6"
    >
      <div
        className="css-go8n81"
      >
        Contiamo
      </div>
      <div
        className="css-11lzgch"
        onClick={undefined}
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23 19a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h4l2-3h6l2 3h4a2 2 0 0 1 2 2z"
          />
          <circle
            cx="12"
            cy="13"
            r="4"
          />
        </svg>
         Gallery
      </div>
      <div
        className="css-11lzgch"
        onClick={undefined}
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.89 1.45l8 4A2 2 0 0 1 22 7.24v9.53a2 2 0 0 1-1.11 1.79l-8 4a2 2 0 0 1-1.79 0l-8-4a2 2 0 0 1-1.1-1.8V7.24a2 2 0 0 1 1.11-1.79l8-4a2 2 0 0 1 1.78 0z"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <polyline
            fill="none"
            points="2.32 6.16 12 11 21.68 6.16"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <line
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            x1="12"
            x2="12"
            y1="22.76"
            y2="11"
          />
        </svg>
         Components
      </div>
      <div
        className="css-1krlqp9"
      />
      <div
        className="css-11lzgch"
        onClick={undefined}
      >
        Logout
      </div>
    </div>
    <div
      className="css-z79p0e"
    >
      <div
        className="css-1u55ha5"
      >
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              Data Entry
              
            </div>
          </div>
          
        </div>
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              Feedback
              
            </div>
          </div>
          
        </div>
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              UI Elements
              
            </div>
          </div>
          
        </div>
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              Navigation
              
            </div>
          </div>
          
        </div>
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              Resources
              
            </div>
          </div>
          
        </div>
        <div
          className="css-pp57kh  "
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="css-5ohi6e header"
              onClick={[Function]}
            >
              Icons
              
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="css-1sjzxut"
      >
        <div
          className="Canvas__body"
        />
      </div>
    </div>
  </div>
</div>
`
